### ISSUE TITLE
Cancel pending operations and unbind listeners on detach

### ISSUE THREAD

**Issue by 0x4d** [renew-js/renew-lib#79]
## Describe the bug
All methods relying on the main container being part of the DOM will lead to a multitude of errors after calling `Viewer.detach()`.

## To Reproduce
1. Use something like the create tool and start creating a shape
2. Detach the current instance
3. The preview isn't cleared and the cursor stays the same 
4. Every `MouseMove` throws an `NS_ERROR_FAILURE`

## Expected behavior
Detaching and reattaching a diagram instance should not lead to any errors.

## Additional context
Use the EventBus to announce an impending detach, so that all modules can react and prepare for it.
